internal/provider/provider_test.go

```diff
@@ -13,6 +14,26 @@ import (
 	"github.com/mongodb/terraform-provider-mongodbatlas/internal/provider"
 )
 
+// TestAutogenResourcesNotEnabled prevents accidental commits including autogenerated resources in the provider.
+func TestAutogenResourcesNotEnabled(t *testing.T) {
```

Comment:
I am in doubt if for this PoC branch we want to leave the generated resources configured, mainly thinking about docs generation as well. We can continue to talk over this, no need to make a call now.